egui_plot/src/items/band.rs

```diff
@@ -0,0 +1,244 @@
+    ///
+    /// Use [`Self::with_series`] to provide data and optionally [`Self::with_color`]
+    /// / [`Self::with_alpha`] to style it.
+    pub fn new(name: &str) -> Self {
+        Self {
+            base: PlotItemBase::new(name.into()),
+            color: Color32::TRANSPARENT,
+            alpha: DEFAULT_ALPHA,
+            xs: Vec::new(),
+            y_min: Vec::new(),
+            y_max: Vec::new(),
+        }
+    }
+
+    /// Override the item's stable id.
+    #[inline]
+    pub fn with_id(mut self, id: impl Into<egui::Id>) -> Self {
+        self.base.id = id.into();
+        self
+    }
+
+    /// Set the base RGB color of the band (alpha is taken from [`Self::with_alpha`]).
+    #[inline]
+    pub fn with_color(mut self, color: Color32) -> Self {
+        self.color = color;
+        self
+    }
+
+    /// Set the alpha component (0..=255) for the band fill.
+    #[inline]
+    pub fn with_alpha(mut self, alpha: u8) -> Self {
+        self.alpha = alpha;
+        self
+    }
+
+    /// Provide series data. All inputs must have identical length.
+    ///
+    /// NaN/non-finite samples are skipped segment-wise during tessellation.
+    pub fn with_series(mut self, xs: &[f64], y_min: &[f64], y_max: &[f64]) -> Self {
+        assert_eq!(
+            xs.len(),
+            y_min.len(),
+            "Band: xs and y_min must have the same length"
+        );
+        assert_eq!(
+            xs.len(),
+            y_max.len(),
+            "Band: xs and y_max must have the same length"
+        );
+
+        self.xs.clear();
+        self.y_min.clear();
+        self.y_max.clear();
+
+        self.xs.extend_from_slice(xs);
+        self.y_min.extend_from_slice(y_min);
+        self.y_max.extend_from_slice(y_max);
+        self
+    }
+
+    /// Compute data bounds for auto-scaling.
+    fn compute_bounds(&self) -> Option<PlotBounds> {
+        if self.xs.is_empty() {
+            return None;
+        }
+
+        let mut min_x = f64::INFINITY;
+        let mut max_x = f64::NEG_INFINITY;
+        let mut min_y = f64::INFINITY;
+        let mut max_y = f64::NEG_INFINITY;
+
+        for i in 0..self.xs.len() {
+            let x = self.xs[i];
+            let yl = self.y_min[i];
+            let yu = self.y_max[i];
+
+            if !(x.is_finite() && yl.is_finite() && yu.is_finite()) {
+                continue;
+            }
+
+            min_x = min_x.min(x);
+            max_x = max_x.max(x);
+
+            let lo = yl.min(yu);
+            let hi = yl.max(yu);
+
+            min_y = min_y.min(lo);
+            max_y = max_y.max(hi);
+        }
+
+        if !(min_x.is_finite() && max_x.is_finite() && min_y.is_finite() && max_y.is_finite()) {
+            return None;
+        }
+
+        Some(PlotBounds::from_min_max([min_x, min_y], [max_x, max_y]))
+    }
+
+    /// Build a filled triangle mesh for the band in screen space.
+    fn build_mesh(&self, transform: &PlotTransform) -> Mesh {
+        let mut mesh = Mesh::default();
```

Comment:
Since we know the size of xs/ys, we could preallocate capacity for Mesh.indices/vertices.